modules/nwtc-library/src/NWTC_IO.f90

```diff
@@ -2263,7 +2263,7 @@ SUBROUTINE DispCompileRuntimeInfo()
          compiled_precision = 'unknown'
       END IF
 
-#if defined(__INTEL_COMPILER) && (__INTEL_COMPILER<1800)
+#ifndef HAS_FORTRAN2008_FEATURES
```

Comment:
I'm a little concerned that a non-Intel compiler could get into this ifndef branch and then have trouble because `__INTEL_COMPILER` isn't defined on the next line.

Reply:
@bjonkman Please see the suggested change I had put in after submitting this pull request.

Reply:
@sayerhs, yes, that helps. However, if it's not an Intel compiler, we could still get into the #else branch here when `!defined(HAS_FORTRAN2008_FEATURES)`, and then `compiler_version()` wouldn't be defined.

Reply:
Pushed another commit which I think covers the different use-cases: 1. we have Fortran2008 features, 2. We don't have F2008 but are using an Intel compiler, 3. neither... so we don't print anything regarding compiler. #454 also had an issue where it would have tried to access `compiler_version` on a non-Intel compiler which might not have had Fortran2008.

Reply:
Resolving this conversation to allow discussion on the newer commit.